hnn_core/gui/_viz_manager.py

```diff
@@ -49,26 +48,84 @@
 ]
 
 fig_templates = {
-    "2row x 1col (1:3)": {
-        "kwargs": "gridspec_kw={\"height_ratios\":[1,3]}",
+    "[Blank] 2row x 1col (1:3)": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 3]}
+        },
         "mosaic": "00\n11",
     },
-    "2row x 1col (1:1)": {
-        "kwargs": "gridspec_kw={\"height_ratios\":[1,1]}",
+    "[Blank] 2row x 1col (1:1)": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 1]}
+        },
         "mosaic": "00\n11",
     },
-    "1row x 2col (1:1)": {
-        "kwargs": "gridspec_kw={\"height_ratios\":[1,1]}",
+    "[Blank] 1row x 2col (1:1)": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 1]}
+        },
         "mosaic": "01\n01",
     },
-    "single figure": {
-        "kwargs": "",
+    "[Blank] single figure": {
+        "kwargs": {
+            "gridspec_kw": ""
+        },
         "mosaic": "00\n00",
     },
-    "2row x 2col (1:1)": {
-        "kwargs": "gridspec_kw={\"height_ratios\":[1,1]}",
+    "[Blank] 2row x 2col (1:1)": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 1]}
+        },
         "mosaic": "01\n23",
+    }
+}
+
+data_templates = {
+    "Drive-Dipole": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 3]}
+        },
+        "mosaic": "00\n11",
+        "ax_plots": [("ax0", "input histogram"), ("ax1", "current dipole")]
+    },
+    "Dipole Layers": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 1, 1]}
+        },
+        "mosaic": "0\n1\n2",
+        "ax_plots": [("ax0", "layer2 dipole"), ("ax1", "layer5 dipole"),
+                     ("ax2", "current dipole")]
+    },
+    "Drive-Spikes": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 3]}
+        },
+        "mosaic": "00\n11",
+        "ax_plots": [("ax0", "input histogram"), ("ax1", "spikes")]
+    },
+    "Dipole-Spectrogram": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 3]}
+        },
+        "mosaic": "00\n11",
+        "ax_plots": [("ax0", "current dipole"), ("ax1", "spectrogram")]
+    },
+    "Drive-Dipole-Spectrogram": {
+        "kwargs": {
+            "gridspec_kw": {"height_ratios": [1, 1, 2]}
+        },
+        "mosaic": "0\n1\n2",
+        "ax_plots": [("ax0", "input histogram"), ("ax1", "current dipole"),
+                     ("ax2", "spectrogram")]
     },
+    "PSD Layers": {
```

Comment:
```suggestion
    "PSD Layers (3 x 1)": {
```

Reply:
I added the subplot grid spec to the layout template name